usaspending_api/recipient/v2/views/list_recipients.py

```diff
@@ -1,59 +1,68 @@
-import logging
 import copy
+import logging
 
+from django.db.models import F, Q
 from rest_framework.response import Response
 from rest_framework.views import APIView
-from django.db.models import F, Q
 
 from usaspending_api.common.cache_decorator import cache_response
 from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
 from usaspending_api.common.validator.pagination import PAGINATION
 from usaspending_api.common.validator.tinyshield import TinyShield
 from usaspending_api.common.validator.utils import update_model_in_list
 from usaspending_api.recipient.models import RecipientProfile
-from usaspending_api.recipient.v2.lookups import SPECIAL_CASES
+from usaspending_api.recipient.v2.lookups import AWARD_TYPES, SPECIAL_CASES
 
 logger = logging.getLogger(__name__)
 
-
-# based from all_award_types_mappings in usaspending_api.awards.v2.lookups.lookups
-AWARD_TYPES = {
-    "contracts": {"amount": "last_12_contracts", "filter": "contract"},
-    "grants": {"amount": "last_12_grants", "filter": "grant"},
-    "direct_payments": {"amount": "last_12_direct_payments", "filter": "direct payment"},
-    "loans": {"amount": "last_12_loans", "filter": "loans"},
-    "other_financial_assistance": {"amount": "last_12_other", "filter": "other"},
-}
+award_types = list(AWARD_TYPES.keys()) + ["all"]
+RECIPIENT_MODELS = [
+    {"name": "keyword", "key": "keyword", "type": "text", "text_type": "search"},
+    {"name": "award_type", "key": "award_type", "type": "enum", "enum_values": award_types, "default": "all"},
+]
 
 
-def get_recipients(filters={}):
-    lower_limit = (filters["page"] - 1) * filters["limit"]
-    upper_limit = filters["page"] * filters["limit"]
-
+def build_duns_base_query(filters):
     qs_filter = Q()
     if "keyword" in filters:
         qs_filter |= Q(recipient_name__contains=filters["keyword"].upper())
         qs_filter |= Q(recipient_unique_id__contains=filters["keyword"])
```

Comment:
A possible optimization here could be to also add in a check for `IS NOT NULL` in djangoese 

iirc, you said this turns into sql that looks like
```SQL
"recipient_profile"."recipient_name"::TEXT LIKE '%DEPARTMENT%'
OR "recipient_profile"."recipient_unique_id"::TEXT LIKE '%department%'
```

Since the vast majority of `recipient_unique_id` values are null, would this go faster if that clause was added?

e.g. to result in

```SQL
WHERE
-- ...
(
    "recipient_profile"."recipient_name" IS NOT NULL
    AND
    "recipient_profile"."recipient_name"::TEXT LIKE '%DEPARTMENT%'
)
OR     
(
    "recipient_profile"."recipient_unique_id" IS NOT NULL
    AND
    "recipient_profile"."recipient_unique_id"::TEXT LIKE '%department%'
)
```

So I guess the question is, when doing a `LIKE` does Postgres already ignore nulls, or if not would dropping rows with null help the `LIKE` run more efficiently. 

I can lookup this question. Unless @tony-sappe has the answer.

```

Reply:
I don't know if documentation will provide the answer. Only trial-and-error experimentation on the exact data and DB hardware will shed light on if adding the checks will help. My assumption is creating partial indexes on non-null values would be more beneficial. That was something that was extensively done on many materialized view indexes.

Reply:
We can defer this but may want to experiment with this style of query directly in sql if we find things still slower or a high degree of writes (e.g. in-memory table scans) seeing under concurrent querying.